src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -401,6 +405,79 @@ protected override void Dispose(bool disposing)
         base.Dispose(disposing);
     }
 
+    private static Sampler ResolveSamplerFromSpecificationConfigurationKeys(IConfiguration configuration, Sampler? stateSampler)
+    {
+        Sampler? sampler = null;
+
+        if (stateSampler != null)
+        {
+            sampler = stateSampler;
+        }
+
+        if (configuration.TryGetStringValue(TracesSamplerConfigKey, out var tracesSampler))
+        {
+            if (sampler != null)
+            {
+                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent(
+                    $"Trace sampler configuration value '{tracesSampler}' has been ignored because a value '{sampler.Description}' was set previously.");
```

Comment:
It's better to log the sampler type than the sampler description.

```suggestion
                    $"Trace sampler configuration value '{tracesSampler}' will be ignored because a sampler of type: `{sampler.GetType().FullName}` was set through code.");
```

Reply:
Would you recommend switching out the Description prop for types in tests and log messages, or just the log messages?

Reply:
Just the log messages. It's good to check the description in unit tests.